mava/components/jax/training/step.py

```diff
@@ -113,6 +137,45 @@ def on_training_step(self, trainer: SystemTrainer) -> None:
         trainer.store.trainer_logger.write({**results})
 
 
+class Step(Component):
+    @abc.abstractmethod
+    def on_training_step_fn(self, trainer: SystemTrainer) -> None:
+        """[summary]"""
+
+    @staticmethod
+    def name() -> str:
+        """_summary_
+
+        Returns:
+            _description_
+        """
+        return "sgd_step"
+
+    @staticmethod
+    def required_components() -> List[Type[Callback]]:
+        """List of other Components required in the system for this Component to function.
+
+        TrainerDataset required for config sample_batch_size.
+        ParallelSequenceAdder required for config sequence_length.
+        BaseTrainerInit required to set up trainer.store.networks
+        and trainer.store.trainer_agent_net_keys.
+        EpochUpdate required to set up trainer.store.epoch_update_fn.
+        MinibatchUpdate required to set up trainer.store.opt_states.
+        Networks required to set up trainer.store.key.
+
+        Returns:
+            List of required component classes.
+        """
+        return [
+            TrainerDataset,
+            mava.components.jax.building.adders.ParallelSequenceAdder,
+            BaseTrainerInit,
+            mava.components.jax.training.model_updating.EpochUpdate,
+            mava.components.jax.training.model_updating.MinibatchUpdate,
+            Networks,
+        ]
```

Comment:
See elsewhere for what I've done when there are specific components of the abstract that require extra dependencies. I can do the same here?

`MinibatchUpdate` and `EpochUpdate` are already the abstract versions.
`ParallelSequenceAdder` is specific, so maybe I should move that dependency to the `MAPGWithTrustRegionStep` class?

Reply:
What you have done elsewhere makes sense. The reason I think the update components should still move is because they are abstract for policy gradient algorithms but a system like dqn might not need them at all. What are yout thoughts?

Reply:
Any `Step` will always need `BaseTrainerInit` and `Networks`, for sure.
Also, it seems like both datasets we have defined a config of `sample_batch_size`, so that's also always needed, no?

Would we not also need an `EpochUpdate` and a `MinibatchUpdate` for DQN?

Reply:
`BaseTrainerInit`, `Networks` and `TrainerDataset` seem general enough to me. But `MinibatchUpdate` and `EpochUpdate` seem a bit too PPO specific to me.

Reply:
Agreed, thanks for the feedback. I've made the requested changes.

Reply:
Thank you :)